base/src/main/java/in/droidcon/base/repo/remote/ServiceFactory.kt

```diff
@@ -0,0 +1,68 @@
+package `in`.droidcon.base.repo.remote
+
+import com.google.gson.FieldNamingPolicy
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+
+object ServiceFactory {
+    /**
+     * Provide "make" methods to create instances of [ServiceClass]
```

Comment:
I'm also not too familiar with Clean Architecture, I followed https://github.com/bufferapp/android-clean-architecture-boilerplate/blob/master/remote/src/main/java/org/buffer/android/boilerplate/remote/BufferooServiceFactory.kt

Not only that, another logic behind this factory is that, we will be using Retrofit in different Modules with different Service classes, and these boilerplates to Koin Module classes will bloat them, where we simply want to have an instance of the Service interface we defined there, so I created the Generic ServiceFactory classes, and in Module classes we simply need to call one method.

Reply:
Not sure I get this completely... Koin supports multi module DI out of the box, so it would be like defining one module with all the necessary things and just using it wherever we want to. 

I see your point of having one method call to setup and run everything though. 
Would it be a bad idea to just have these object creations in the koin Module class and use it directly ? I like the idea of having abstraction over the methods but in this case since koin will inject them for us I am not sure if having a generic abstraction class helps with anything...

@Hariofspades - Thoughts ?

Reply:
I think I wasn't able to express myself well enough, so sorry for that.
My objective behind this file is as below, also please correct me if I'm wrong.
Right now there's only one Android module (`speakers`) using retrofit, so it's perfectly ok to move these boilerplates to the `KoinModule` class in that module. But we are going to have ~3 more different Android modules, all of them are expected to use retrofit. So, in that scenario, if we keep the retrofit boilerplates in particular modules inside Koin, then we will be ending up having to copy paste these boilerplates in each `KoinModule` class inside each of the Android modules, resulting in redundant code, the only thing unique would be the `ServiceClass` for each module.
We also can't move it inside Koin of the base module, since the `ServiceClass` interface should be defined by each module separately.
Another approach can be to inject the object of retrofit through base module and create the `ServiceClass` instance inside `KoinModule` of different modules, but why inject Retrofit instance and block the memory?

That's the reason I created this methods, so that `ServiceClass` instance can be created in `KoinModule` of different Android modules easily without any redundancies/boilerplates.
@adnan-SM @Hariofspades @vivek1794